<launch>

  <!-- usb_cam -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/videoCAM" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="autoexposure" value="false" />
    <param name="exposure" value="26" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name="reprojection" pkg="yolov3_trt_ros" type="Reprojection.py" output="screen">
    <param name="scan_topic" value="/scan" />
    <param name="image_topic" value="/usb_cam/image_raw" />
    <param name="calib_file" value="$(find yolov3_trt_ros)/data/calibration_result.txt" />
    <param name="config_file" value="$(find yolov3_trt_ros)/config/config.yaml" />
    <param name="laser_point_radius" type="int" value="3" />    <!-- Reprojected laser point radius size in pixels -->
    <param name="time_diff" type="double" value="1" />  <!-- Maximum time difference in seconds between image and laser scan to be registered and processed -->
  </node>

  <include file="$(find xycar_lidar)/launch/lidar_noviewer.launch" />

  <node name="yolov3_trt_ros" pkg="yolov3_trt_ros" type="trt_detection.py" output="screen" />
  <node name="trt_driver" pkg="yolov3_trt_ros" type="trt_drive.py" output="screen" />

  <include file="$(find LaneKeepingSystem)/launch/drive.launch"/>
  <param name="config_path" type="str" value="$(find LaneKeepingSystem)/config/config.yaml"/>

  <!-- <node name="rviz" type="rviz" pkg="rviz" args="-d $(find yolov3_trt_ros)/rviz/reprojection.rviz"/> -->
  
</launch>
